components/automate-deployment/pkg/majorupgradechecklist/v3.go

```diff
@@ -253,19 +249,58 @@ func backupCheck() Checklist {
 	}
 }
 
-func diskSpaceCheck() Checklist {
+func diskSpaceCheck(version string, skipDiskSpaceCheck bool, osDestDataDir string) Checklist {
 	return Checklist{
 		Name:        "disk_space_acceptance",
 		Description: "confirmation check for disk space",
 		TestFunc: func(h ChecklistHelper) error {
-			resp, err := h.Writer.Confirm("Ensure you have more than 60 percent free disk space")
+			osPath := getHabRootPath(habrootcmd)
+			habDirSize, err := cm.CalDirSizeInGB(osPath)
+
+			if err != nil {
+				h.Writer.Error(err.Error())
+				return status.Errorf(status.UnknownError, err.Error())
+			}
+			var spaceAvailable bool
+			version, _ = GetMajorVersion(version)
+			var dbDataPath string
+			switch version {
+			case "3":
+				dbDataPath = osPath + "svc/automate-postgresql/data/pgdata"
+			case "4":
+				dbDataPath = osPath + "svc/automate-elasticsearch/data"
+			}
+
+			dbDataSize, err := cm.CalDirSizeInGB(dbDataPath)
+			if err != nil {
+				h.Writer.Error(err.Error())
+				return status.Errorf(status.UnknownError, err.Error())
+			}
+
+			minReqDiskSpace := math.Max(MIN_DIRSIZE_GB, math.Max(habDirSize, dbDataSize)) * 11 / 10
+
+			resp, err := h.Writer.Confirm(fmt.Sprintf("Ensure you have more than %.2f GB of free disk space", minReqDiskSpace))
 			if err != nil {
 				h.Writer.Error(err.Error())
 				return status.Errorf(status.InvalidCommandArgsError, err.Error())
 			}
+			if !skipDiskSpaceCheck {
+				destDir := osPath
+				if osDestDataDir != "" {
+					destDir = osDestDataDir
+				}
+				h.Writer.Printf("Destination directory chosen to check free disk space: %s\n", destDir)
+				h.Writer.Println("To change destination directory please use --os-dest-data-dir flag")
+				spaceAvailable, err = cm.CheckSpaceAvailability(destDir, minReqDiskSpace)
+				if err != nil {
```

Comment:
Both if blocks are returning same thing. I think it would be better to club them in one if block.